src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -556,12 +556,211 @@ codet java_bytecode_convert_methodt::get_array_bounds_check(
+
+ Purpose: Throws NegativeArraySizeException when necessary
+
+\*******************************************************************/
+
+codet java_bytecode_convert_methodt::throw_array_length_exception(
+  const exprt &length,
+  const source_locationt &original_loc)
+{
+  symbolt exc_symbol;
+  if(!symbol_table.has_symbol("NegativeArraySizeException"))
+  {
+    exc_symbol.is_static_lifetime=true;
+    exc_symbol.base_name="NegativeArraySizeException";
+    exc_symbol.name="NegativeArraySizeException";
+    exc_symbol.mode=ID_java;
+    exc_symbol.type=typet(ID_pointer, empty_typet());
+    symbol_table.add(exc_symbol);
+  }
+  else
+    exc_symbol=symbol_table.lookup("NegativeArraySizeException");
+
+  const symbol_exprt &exc=exc_symbol.symbol_expr();
+  side_effect_expr_throwt throw_expr;
+  throw_expr.add_source_location()=original_loc;
+  throw_expr.copy_to_operands(exc);
+
+  const constant_exprt &zero=from_integer(0, java_int_type());
+  const binary_relation_exprt ge_zero(length, ID_ge, zero);
+
+  code_ifthenelset if_code;
+  if_code.add_source_location()=original_loc;
+  if_code.cond()=ge_zero;
+  if_code.then_case()=code_expressiont(throw_expr);
+
+  return if_code;
+}
+
+/*******************************************************************\
+
+Function: throw_null_dereference_exception
+
+  Inputs: 
+
+ Outputs: 
+
+ Purpose: Checks whether expr is null, and throws
+          NullPointerException if necessary
+
+\*******************************************************************/
+
+codet java_bytecode_convert_methodt::throw_null_dereference_exception(
+  const exprt &expr,
+  const source_locationt &original_loc)
+{
+  symbolt exc_symbol;
+  if(!symbol_table.has_symbol("NullPointerException"))
+  {
+    exc_symbol.is_static_lifetime=true;
+    exc_symbol.base_name="NullPointerException";
+    exc_symbol.name="NullPointerException";
+    exc_symbol.mode=ID_java;
+    exc_symbol.type=typet(ID_pointer, empty_typet());
+    symbol_table.add(exc_symbol);
+  }
+  else
+    exc_symbol=symbol_table.lookup("NullPointerException");
+
+  const symbol_exprt &exc=exc_symbol.symbol_expr();
+  side_effect_expr_throwt throw_expr;
+  throw_expr.add_source_location()=original_loc;
+  throw_expr.copy_to_operands(exc);
+
+  const equal_exprt equal_expr(
+    expr,
+    null_pointer_exprt(pointer_typet(empty_typet())));
+  code_ifthenelset if_code;
+  if_code.add_source_location()=original_loc;
+  if_code.cond()=equal_expr;
+  if_code.then_case()=code_expressiont(throw_expr);
+
+  return if_code;
+}
+
+/*******************************************************************\
+
+Function: throw_class_cast_exception
+
+  Inputs: 
+
+ Outputs: 
+
+ Purpose: Throws ClassCastException when  necessary
+
+\*******************************************************************/
+
+codet java_bytecode_convert_methodt::throw_class_cast_exception(
+  const exprt &class1,
+  const exprt &class2,
+  const source_locationt &original_loc)
+{
+  // TODO: use std::move
```

Comment:
To avoid having to fix this in 4 places - could a helper function `get_exception(const irep_idt &exception_type)` handle this logic?